interactive-etl/setup.sh

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env bash
+
+set -u 
+set -e
+set -o errexit
+
+NAMESPACE=${1:-flink}
+
+printf "\n\nInstalling example components into namespace: %s\n\n" "$NAMESPACE"
+
+# Install CertManager - this is needed by the Flink Kubernetes Operator
+printf "\nChecking for CertManager install\n"
+if kubectl get namespace cert-manager ; then
+    printf "CertManager is already installed\n"
+else
+    kubectl create -f https://github.com/jetstack/cert-manager/releases/download/v1.8.2/cert-manager.yaml
+fi
+
+# Add the Flink operator's helm repo
+helm repo add flink-operator-repo https://downloads.apache.org/flink/flink-kubernetes-operator-1.9.0/
+
+printf "\nChecking for %s namespace\n" "$NAMESPACE"
+if kubectl get namespace "$NAMESPACE" ; then
+    echo "$NAMESPACE namespace already exists"
+else
+    kubectl create namespace "$NAMESPACE"
+fi
+
+printf "\nWaiting for cert-manager webhook to be ready..."
+sleep 1m
+kubectl -n cert-manager wait --for=condition=Available --timeout=60s deployment cert-manager-webhook
```

Comment:
This is effectively a 2m wait so why not set timeout to 120s and let it return instantly for the case where cert manager was already installed?

Reply:
Though on the subjects of waits I was pondering if it was better to have multiple shorter waits in a in loop. Its probably more efficient to just give wait a longer timeout than loop but it means the scripts appear to hang ...